teps/0147-tekton-artifacts-phase1.md

```diff
@@ -0,0 +1,285 @@
+
+This TEP proposes to define the Tekton Artifacts Provenance data structure that `Steps` can output (Phase 1). It also shows how this information is carried into the `TaskRun Status`. Furthermore, it also proposes how Tekton Chains could parse the `uri` and insert the provenance data into appropriate sections of the `SLSA` Provenance payload.
+
+## Motivation
+
+As defined in the [feature request](https://github.com/tektoncd/pipeline/issues/6326), the reporing of metadata of artifacts in Tekton has grown organically and is redundant and brittle. The structure of results make extremely challenging for a remote Task to produce artifacts of unknown length (e.g. a `ko Task` that produces images from source code does not know beforehand how many images will be produced). There is a need for users to be able to surface provenance output without being limited by the result spec.
+
+This works aims to overcome these limitations.
+
+### Goals
+#### Phase 1
+
+- Provide a mechanism for surfacing artifact provenance data to the `TaskRun` status.
+- Improve the current mechanism of collecting provenance by Tekton Chains that relies on type hinting.
+
+
+#### Phase 2
+- Provide a mechanism for making the artifact provenance data of a `TaskRun` available to downstream `TaskRuns` through variables.
+- Provide an API for users to declare upfront artifacts consumed and produced by `Tasks`.
+- Provide a set of upload and download `StepActions` for different storages.
+- Contribute to the chain of trust by allowing consumer `Tasks` to trust artifacts on a `Workspace` from producer `Tasks`, as long as `TaskRun` status can be trusted.
+- Introduce trusted `StepActions`.
+
+### Non-Goals
+
+- an out-of-the-box mechanism to safely share artifacts and their provenance data between `Tasks` through a reference implementation of the steps required, automatic injection of the steps and provisioning of the required storage.
+- Provide a mechanism for Tekton to inject user-defined digest, verify, upload and download steps, when the artifact API is used. 
+- This proposal does not discuss how to inject or expose artifacts as inputs/outputs of a standalone `TaskRun` and for a `PipelineRun`, even though it sets foundations that could be used to achieve that.
+
+### Use Cases
+
+This TEP contributes to the achievements of the following use cases:
+
+- Achieve SLSA L3 supply chain security through Tekton `PipelineRuns`
```

Comment:
To be clear, this helps towards this goal but it doesn't actually achieves it completely. Correct?

Even if a PipelineRun uses this feature to pass content from one Task to another (e.g. build Task uses output of clone Task), we still need a mechanism to verify the content was not modified during the length of execution of the consumer Task. For instance, the consumer Task first needs to copy input to a volume no other Task has access to (e.g. EmptyDir), and verify the digest of the input. This, I believe, is outside the scope of the this TEP.

Reply:
Yes, you are correct. I think this is a better use case for phase 2. Where we enable passing artifacts between Tasks and rely on `spire` support for non-falsifiable provenance.